examples/lie_labs.py

```diff
@@ -0,0 +1,121 @@
+import torch
+
+import theseus.labs.lie as lie
+import theseus.labs.lie.functional as lieF
+
+batch_size = 5
+
+# ### Lie Tensor creation functions
+g1 = lie.rand(batch_size, lie.SE3, requires_grad=True)
+print(f"Created SE3 tensor with shape {g1.shape}")
+g2 = g1.clone()
+
+# Can create from a tensor as long as it's consistent with the desired ltype
+g3_data = lieF.so3.rand(5)
```

Comment:
Is `g3_data` a tensor of shape (5, 3, 3)? I assume that data for the lie group is the transformation matrix, but just wanted to be sure.

Reply:
In this case, yes. Currently we are only using matrix representation, but ideally, we'll get to the point where we can offer alternative storage options (as you are suggesting above). 

My current vision is that something like `print(g)` would return stuff like:

`LieTensor(..., shape(1, 2, 3), ltype=se3, storage="mat")` # raw tensor is shape (1, 2, 3, 3, 4)
or
`LieTensor(..., shape(1, 2, 3), ltype=se3, storage="q")`  # raw tensor is shape (1, 2, 3, 7)

with a bunch of backend function to interact across representations.

Reply:
That sounds great, would be even nicer if these backend functions are available from the frontend too. (Maybe through lieF?)

Reply:
There has been some discussion about whether to make `lieF` available or not. I don't remember what the cons we discussed were, maybe we had some concerns that people would be confused about what the canonical entry point for the library would be? @mhmukadam do you remember?

Reply:
I think it was mostly that now we have worked out subclassing torch tensor cleanly so exposing two types of API seemed bloat.

If these backend functions are also exposed in the frontend through the class etc then do you think there is still utility in having access to the functional API through lieF directly?